src/redux/features/votes/votesAsyncThunks.ts

```diff
@@ -0,0 +1,43 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { Response2 } from 'src/services/HttpService';
+import VotesService from 'src/services/VotesService';
+import strings from 'src/strings';
+import {
+  DeleteProjectVotesRequestPayload,
+  DeleteProjectVotesResponsePayload,
+  UpsertProjectVotesRequestPayload,
+  UpsertProjectVotesResponsePayload,
+} from 'src/types/Votes';
+
+export const requestProjectVotesUpdate = createAsyncThunk(
+  'votes/update',
+  async (request: { projectId: number; payload: UpsertProjectVotesRequestPayload }, { rejectWithValue }) => {
+    const response: Response2<UpsertProjectVotesResponsePayload> = await VotesService.setProjectVotes(
+      request.projectId,
+      request.payload
+    );
+
+    if (response !== null && response.requestSucceeded) {
```

Comment:
Response will not be null, you should get a valid response object. The service layer catches errors and set the status in requestSucceeded / error properties.